src/vivarium_gates_mncnh/components/antenatal_care.py

```diff
@@ -402,61 +219,46 @@ def get_standard_ultrasound_probability(index: pd.Index) -> pd.Series:
                 standard_probability = ANC_RATES.ULTRASOUND_TYPE[self.location][
                     ULTRASOUND_TYPES.STANDARD
                 ]
-            return pd.Series(standard_probability, index=index)
+            return standard_probability
 
-        def get_ai_assisted_ultrasound_probability(index: pd.Index) -> pd.Series:
-            return 1 - get_standard_ultrasound_probability(index)
-
-        state_A.add_transition(
-            output_state=gets_ultrasound,
-            probability_function=get_ultrasound_probability,
-        )
-        state_A.add_transition(
-            output_state=end_state, probability_function=get_residual_ultrasound_probability
-        )
-        state_B.add_transition(
-            output_state=gets_ultrasound,
-            probability_function=get_ultrasound_probability,
-        )
-        state_B.add_transition(
-            output_state=end_state,
-            probability_function=get_residual_ultrasound_probability,
-        )
-        state_C.add_transition(
-            output_state=gets_ultrasound,
-            probability_function=get_ultrasound_probability,
-        )
-        state_C.add_transition(
-            output_state=end_state,
-            probability_function=get_residual_ultrasound_probability,
-        )
-        state_D.add_transition(output_state=end_state)
+        pop = self.population_view.get(event.index)
+        anc_pop = pop.loc[pop[COLUMNS.ANC_ATTENDANCE] != ANC_ATTENDANCE_TYPES.NONE]
+
+        # determine who gets ultrasound
+        ultrasound_probability = get_ultrasound_probability()
+        gets_ultrasound = self.randomness.choice(
+            anc_pop.index,
+            choices=[True, False],
+            p=[ultrasound_probability, 1 - ultrasound_probability],
+            additional_key="gets_ultrasound",
+        )
+
+        # determine type of ultrasound for those who get it
+        standard_ultrasound_probability = get_standard_ultrasound_probability()
+        ultrasound_pop = anc_pop.loc[gets_ultrasound]
+        ultrasound_type = self.randomness.choice(
+            ultrasound_pop.index,
+            choices=[ULTRASOUND_TYPES.STANDARD, ULTRASOUND_TYPES.AI_ASSISTED],
+            p=[standard_ultrasound_probability, 1 - standard_ultrasound_probability],
+            additional_key="ultrasound_type",
+        )
+
+        pop[COLUMNS.ULTRASOUND_TYPE] = ULTRASOUND_TYPES.NO_ULTRASOUND
+        pop.loc[ultrasound_pop.index, COLUMNS.ULTRASOUND_TYPE] = ultrasound_type
```

Comment:
I dont know if I have just been looking at too many population updates by why do you not need to update the population view here?